src/client.rs

```diff
@@ -178,7 +183,10 @@ where
 
         self.tasks.push(
             async move {
-                let res = store.put_many_keyed(blocks.into_iter()).await;
+                let res = store
+                    .put_many_keyed(blocks.clone().into_iter())
```

Comment:
```suggestion
                    .put_many_keyed(blocks.into_iter())
```

Reply:
I don't think we can drop this clone. If you notice [`map` ](https://github.com/eigerco/beetswap/pull/27/files/38ac7017d6a4e2d4a99f828e4f76709d83ccfcb7#diff-7f93c4e263c4e9ec748f804c7fd04a3b2fde86ffd741fb5516d67e1097bae4c1R189), changes the `Ok` result to contain the inserted blocks, which is then forwarded in `TaskResult`.